skrub/_multi_agg_joiner.py

```diff
@@ -0,0 +1,417 @@
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from skrub import _join_utils
+from skrub._agg_joiner import AggJoiner
+from skrub._dataframe._namespace import is_pandas, is_polars
+from skrub._utils import _is_array_like, atleast_2d_or_none
+
+
+class MultiAggJoiner(BaseEstimator, TransformerMixin):
+    """Extension of the AggJoiner to multiple auxiliary tables.
+
+    Apply numerical and categorical aggregation operations on the `cols`
+    to aggregate, selected by dtypes. See the list of supported operations
+    at the parameter `operations`.
+
+    If `cols` is not provided, `cols` are all columns from `aux_tables`,
+    except `aux_keys`.
+
+    As opposed to AggJoiner, all parameters here must be iterable of str,
+    or iterable of iterable of str. If a single table is fed to the MultiAggJoiner,
+    a similar behavior to the AggJoiner can be obtained by transforming all parameters
+    to a list of parameters, i.e. ``key = "my_key"`` becomes ``key = ["my_key"]``.
+
+    Accepts :obj:`pandas.DataFrame` and :class:`polars.DataFrame` inputs.
+
+    Parameters
+    ----------
+    aux_tables : iterable of DataFrameLike or "X"
+        Auxiliary dataframes to aggregate then join on the base table.
+        The placeholder string "X" can be provided to perform
+        self-aggregation on the input data. To provide a single auxiliary table,
+        ``aux_tables = [table]`` is supported, but not ``aux_tables = table``.
+        It's possible to provide both the placeholder "X" and auxiliary tables,
+        as in ``aux_tables = [table, "X"]``. If that's the case, the second table will
+        be replaced by the input data.
+
+    keys : iterable of str, or iterable of iterable of str, default=None
+        The column names to use for both `main_keys` and `aux_key` when they
+        are the same. Provide either `key` or both `main_keys` and `aux_keys`.
+        If there are multiple auxiliary tables, `keys` will be used to join all
+        of them.
+
+    main_keys : iterable of str, or iterable of iterable of str, default=None
+        Select the columns from the main table to use as keys during
+        the join operation.
+        If `main_keys` contains multiple columns, we will perform a multi-column join.
+
+    aux_keys : iterable of str, or iterable of iterable of str, default=None
+        Select the columns from the auxiliary dataframes to use as keys during
+        the join operation. Note that ``[["a"], ["b"], ["c", "d"]]`` is a valid input
+        while ``["a", "b", ["c", "d"]]`` is not.
+
+    cols : iterable of str, or iterable of iterable of str, default=None
+        Select the columns from the auxiliary dataframes to use as values during
+        the aggregation operations.
+
+        If set to `None`, `cols` are all columns from table, except `aux_keys`.
+
+    operations : iterable of str, or iterable of iterable of str, default=None
```

Comment:
I wonder if we could say that the same operations are applied to all tables, ie operations here has the same type as in aggjoiner.

I don't see why we would want to specify operations specific to each table but not specific to each column within a table

Reply:
I agree that we could select columns to apply operations on, and we would need to rfc the agg joiner for this

Reply:
maybe keep things simple for now and do the same operation for everybody (besides the continuous vs categorical split), and then we can think if we want to allow specifying operations by column or if that use case is too specific / covered well enough by chaining several aggjoiners